objects/Fan-MadeAccessories.aa8b38/DisplacementTool.0f1374.ttslua

```diff
@@ -1,62 +1,70 @@
 
 local excluded = {
   ["b7b45b"] = true,
   ["f182ee"] = true,
   ["721ba2"] = true
 }
 
----@type table<string, table<string, any>>
 local OFFSETS = {
-  left = {
-    axis = "z",
-    offset = 7.67
-  },
-  right = {
-    axis = "z",
-    offset = -7.67
-  },
-  up = {
-    axis = "x",
-    offset = 6.54
-  },
-  down = {
-    axis = "x",
-    offset = -6.54
-  }
+  left  = { x = 0.00, y = 0, z = 7.67 },
+  right = { x = 0.00, y = 0, z = -7.67 },
+  up    = { x = 6.54, y = 0, z = 0.00 },
+  down  = { x = -6.54, y = 0, z = 0.00 }
 }
 
----@param player Player
----@param direction string
-function shift(player, direction)
-  if self:getDescription()=="" then
-    player:broadcast("The description of the displacement tool must contain (only) the GUID for a scripting zone that contains the play field.", Color.RED)
-    return
-  end
+local UI_offset = 1.15
+
+local buttonParamaters          = {}
+buttonParamaters.function_owner = self
+buttonParamaters.label          = ""
+buttonParamaters.height         = 500
+buttonParamaters.width          = 500
+buttonParamaters.color          = { 0, 0, 0, 0 }
 
-  local zone = getObjectFromGUID(self:getDescription())
-  local offset = OFFSETS[direction]
+function onLoad()
+  -- index 0: left
+  buttonParamaters.click_function = "shift_left"
+  buttonParamaters.tooltip        = "Move left"
+  buttonParamaters.position       = { -UI_offset, 0, 0 }
+  self.createButton(buttonParamaters)
 
+  -- index 1: right
+  buttonParamaters.click_function = "shift_right"
+  buttonParamaters.tooltip        = "Move right"
+  buttonParamaters.position       = { UI_offset, 0, 0 }
+  self.createButton(buttonParamaters)
+
+  -- index 2: up
+  buttonParamaters.click_function = "shift_up"
+  buttonParamaters.tooltip        = "Move up"
+  buttonParamaters.position       = { 0, 0, -UI_offset }
+  self.createButton(buttonParamaters)
+
+  -- index 3: down
+  buttonParamaters.click_function = "shift_down"
+  buttonParamaters.tooltip        = "Move down"
+  buttonParamaters.position       = { 0, 0, UI_offset }
+  self.createButton(buttonParamaters)
+end
+
+function shift_left(color) shift(color, "left") end
+
+function shift_right(color) shift(color, "right") end
+
+function shift_up(color) shift(color, "up") end
+
+function shift_down(color) shift(color, "down") end
+
+function shift(color, direction)
   if not zone then
-    player:broadcast("The description of the displacement tool either contains text that is not a GUID or contains a GUID for a scripting zone that does not exist.", Color.RED)
+    broadcastToColor("Scripting zone couldn't be found.", color, "Red")
     return
   end
 
-  local adjustment = Vector(0, 0, 0)
-  adjustment[offset.axis] = offset.offset
-
-  ---@param object TTSObject
-  for _, object in ipairs(zone:getObjects()) do
-    if not (excluded[object:getGUID():lower()] or object:hasTag("displacement_excluded")) then
-      object:translate(adjustment)
+  for _, object in ipairs(zone.getObjects()) do
```

Comment:
From an allocation of responsibility perspective, I think it would make sense for the move code to be owned by Playarea and accessed via API.

Reply:
As talked about, that will be done when you add the PlayareaAPI during the locations connection script.